### PR TITLE
SCC-666

### PR DESCRIPTION
**Pull request by owen-opencast**
Can test by running with SCC-666 branch of stub and updates to service manager, then logging in with AA020513A. When clicking on change main address, should be redirected to Tax Credits Service. When clicking on change postal address, should go to our change of postal address journey.

### CODE REVIEW

File: app/connectors/TaxCreditsConnector.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        s"$taxCreditsUrl/tcs/$nino/dashboard-data"
+      ) recoverWith errorHandling
+    )
+
+  private def errorHandling[T]: PartialFunction[Throwable, Future[T]] = {
+    case validationError: JsValidationException =>
+      logger.warn(validationError.getMessage())
+      Future.failed(validationError)
+    case notFound: NotFoundException            =>
+      Future.failed(new NotFoundException(notFound.getMessage))
```
Comment (ccatosdevelopment) [inline app/connectors/TaxCreditsConnector.scala:52]: Should there be a logger.warn line here as well?

File: test/resources/dashboard-data.json [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    "address": {
+      "addressLine1": "18 Road Street",
+      "addressLine2": "Hexham",
+      "addressLine3": "Northumberland",
+      "postCode": "NE43 8DZ"
+    },
+    "rls": false,
+    "wtcPaymentFrequency": "4 WEEKLY",
+    "ctcPaymentFrequency": "4 WEEKLY",
+    "dayPhoneNumber": "02903 9392 932",
```
Comment (ccatosdevelopment) [inline test/resources/dashboard-data.json:15]: Is this a UK phone number? If so, should it be 1 digit fewer?

File: test/resources/dashboard-data.json [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  "awardDetails": {
+    "applicationId": "000000000005678",
+    "isJoint": false,
+    "claimantNumber": 1,
+    "mainApplicantNino": "AA000003A",
+    "availableForCOCAutomation": false,
+    "renewalsJourney": false,
+    "additionalCocInformation": {
+      "contactInformation": {
+        "primaryApplicantRegularNumber": "02903 9392 932"
```
Comment (ccatosdevelopment) [inline test/resources/dashboard-data.json:108]: (as above) Is this a UK phone number? If so, should it be 1 digit fewer?

File: app/connectors/TaxCreditsConnector.scala (hunk 1 above)
Comment (owen-opencast) [reply app/connectors/TaxCreditsConnector.scala:52]: No, NotFound is returned when a user doesn't not have tax credits - as it's actually the most likely state, and expected, we don't need a warning

File: test/resources/dashboard-data.json (hunk 2 above)
Comment (owen-opencast) [reply test/resources/dashboard-data.json:15]: Not sure! I copied the data from somewhere else, but in any case it isn't used - the user is determined to have tax credits if an Ok response is returned from tax-credits-broker, the content doesn't really matter

File: test/resources/dashboard-data.json (hunk 3 above)
Comment (owen-opencast) [reply test/resources/dashboard-data.json:108]: As above

**Pull request merged**